tools/NetCorePsd1Sync/NetCorePsd1Sync/NetCoreDefinitionGenerator.cs

```diff
@@ -0,0 +1,141 @@
+            var nameLinePrefixIndex = nameLine.IndexOf(ElementPrefix, StringComparison.Ordinal);
+            var name = nameLine.Substring(nameLinePrefixIndex + ElementPrefix.Length, nameLine.LastIndexOf(ElementPostfix, StringComparison.Ordinal) - (nameLinePrefixIndex + ElementPrefix.Length));
+
+            var authorDisplayName = AttributeHelper.GetPropertyAttributeValue<PsDefinitionHeader, string, DisplayNameAttribute, string>(pd => pd.Author, attr => attr.DisplayName, String.Empty);
+            var authorLine = lines.First(l => l.Contains(authorDisplayName));
+            var authorLineDelimiterIndex = authorLine.IndexOf(HeaderDelimiter, StringComparison.Ordinal);
+            var author = authorLine.Substring(authorLineDelimiterIndex + HeaderDelimiter.Length);
+
+            var dateDisplayName = AttributeHelper.GetPropertyAttributeValue<PsDefinitionHeader, DateTime, DisplayNameAttribute, string>(pd => pd.Date, attr => attr.DisplayName, String.Empty);
+            var dateLine = lines.First(l => l.Contains(dateDisplayName));
+            var dateLineDelimiterIndex = dateLine.IndexOf(HeaderDelimiter, StringComparison.Ordinal);
+            var date = dateLine.Substring(dateLineDelimiterIndex + HeaderDelimiter.Length);
+
+            return new PsDefinitionHeader
+            {
+                ModuleName = name,
+                Author = author,
+                Date = DateTime.Parse(date)
+            };
+        }
+
+        public static Hashtable GetHashtable(PowerShell powershell, string path)
+        {
+            var script = $"Get-Content '{path}' | Out-String | Invoke-Expression";
+            powershell.AddScript(script);
+            if (!(powershell.Invoke().First().BaseObject is Hashtable hashtable))
+            {
+                throw new FileFormatException("Missing manifest hashtable data");
+            }
+            hashtable.Add("FilePath", path);
+            return hashtable;
+        }
+
+        public static IEnumerable<Hashtable> GetDesktopHashtables(IEnumerable<string> desktopFilePaths)
+        {
+            using (var powershell = PowerShell.Create())
+            {
+                foreach (var path in desktopFilePaths)
+                {
+                    yield return GetHashtable(powershell, path);
+                }
+            }
+        }
+
+        public static PsDefinition CreateNetCoreDefinition(Hashtable desktopData)
+        {
+            var psData = new PsData();
+            var desktopPsData = desktopData.GetValueAsHashtable("PrivateData").GetValueAsHashtable("PSData");
+            if (desktopPsData.Any())
+            {
+                psData = new PsData
+                {
+                    Tags = desktopPsData.GetValueAsStringList("Tags"),
+                    LicenseUri = new Uri(desktopPsData.GetValueAsString("LicenseUri")),
+                    ProjectUri = new Uri(desktopPsData.GetValueAsString("ProjectUri")),
+                    ReleaseNotes = String.Empty
```

Comment:
Do we need to represent 'Prerelease' here?

Reply:
Great find! I had `Prerelease` in the model, but forgot to include it in the generation logic. Fixed this.